From 24739afc3e8549f7361dd2ac7f8f5368f8fe0f79 Mon Sep 17 00:00:00 2001
From: Jerome Brunet <jbrunet@baylibre.com>
Date: Thu, 30 Mar 2017 16:06:10 +0200
Subject: [PATCH 45/79] WIP: ARM64: dts: meson-gxl-p230: add sound card support

This commit add sound card support to the meson-gxbb-p230
his patch is not intended to be merged upstream. The architecture of
the sound card has not been determined yet. To handle the routing caps
of meson SoCs, we will probably have to make our own card.

This implementation using simple-card is provided for testing purpose only

Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>
---
 .../arm64/boot/dts/amlogic/meson-gxl-s905d-p230.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-p230.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-p230.dts
index 770a332..5da5206 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-p230.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-p230.dts
@@ -90,6 +90,27 @@
 		compatible = "linux,spdif-dit";
 		status = "okay";
 	};
+
+	soc {
+		sound {
+			compatible = "simple-audio-card";
+			simple-audio-card,name = "meson-gxl-p230";
+
+			simple-audio-card,dai-link@0 {
+				plat {
+					sound-dai = <&aiu_spdif_dma>;
+				};
+
+				cpu {
+					sound-dai = <&spdif_dai>;
+				};
+
+				codec {
+					sound-dai = <&spdif_out>;
+				};
+			};
+		};
+	};
 };
 
 &cec_AO {
-- 
1.9.1

